src/overlays/actors/ovl_Oceff_Storm/z_oceff_storm.c

```diff
@@ -33,22 +31,183 @@ const ActorInit Oceff_Storm_InitVars = {
     (ActorFunc)OceffStorm_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_8098176C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981928.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981B48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981BB8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Draw.s")
+void OceffStorm_SetupAction(OceffStorm* this, OceffStormActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+s32 func_8098176C(PlayState* play) {
+    s32 phi_v1 = 0;
```

Comment:
```suggestion
    s32 phi_v1 = false;
```
looks boolean, applies below as well.